oonib/testhelpers/http_helpers.py

```diff
@@ -1,16 +1,87 @@
+import os
 import json
 import random
 import string
+import tempfile
+from hashlib import sha256
 
-from twisted.internet import protocol, defer
+from urlparse import urlparse
 
-from cyclone.web import RequestHandler, Application
+from twisted.python.failure import Failure
+
+from twisted.internet.protocol import Factory, Protocol
+from twisted.internet.endpoints import TCP4ClientEndpoint
+
+from twisted.internet.error import DNSLookupError, TimeoutError
+from twisted.internet.error import ConnectionRefusedError
+from twisted.internet import protocol, defer, reactor
+
+from twisted.names.error import DNSNameError, DNSServerError
+from twisted.names import client as dns_client
+from twisted.names import dns
+
+from twisted.web._newclient import HTTPClientParser, ParseError
+from twisted.web.client import Agent, BrowserLikeRedirectAgent, readBody
+from twisted.web.client import ContentDecoderAgent, GzipDecoder
+from twisted.web.http_headers import Headers
+from twisted.web import error
+
+from twisted.web._newclient import ResponseFailed
+
+from cyclone.web import RequestHandler, Application, HTTPError
+from cyclone.web import asynchronous
 
 from twisted.protocols import policies, basic
 from twisted.web.http import Request
 
 from oonib import log, randomStr
 
+# Monkey patch to overcome issue in parsing of HTTP responses that don't
+# contain 3 parts in the HTTP response line
```

Comment:
[In a recent RFC](https://tools.ietf.org/html/rfc7230#section-3.1.2) the grammar for the response line is:

```
status-line = HTTP-version SP status-code SP reason-phrase CRLF
reason-phrase  = *( HTAB / SP / VCHAR / obs-text )
```

Unless the `*` has a very surprising meaning, I'd say it would be theoretically possible to omit the reason phrase, but I'd also say an implementation SHOULD NOT